sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/RemoteDependencyData.cs

```diff
@@ -102,5 +78,59 @@ public RemoteDependencyData(int version, Activity activity, ref ActivityTagsProc
             TraceHelper.AddActivityLinksToProperties(activity, ref activityTagsProcessor.UnMappedTags);
             TraceHelper.AddPropertiesToTelemetry(Properties, ref activityTagsProcessor.UnMappedTags);
         }
+
+        private void SetHttpDependencyPropertiesAndDependencyName(Activity activity, ref AzMonList httpTagObjects, bool isNewSchemaVersion, out string dependencyName)
+        {
+            string? httpUrl;
+            if (isNewSchemaVersion)
+            {
+                httpUrl = AzMonList.GetTagValue(ref httpTagObjects, SemanticConventions.AttributeUrlFull)?.ToString();
+                dependencyName = httpTagObjects.GetNewSchemaHttpDependencyName(httpUrl) ?? activity.DisplayName;
+            }
+            else
+            {
+                httpUrl = httpTagObjects.GetDependencyUrl();
+                dependencyName = httpTagObjects.GetHttpDependencyName(httpUrl) ?? activity.DisplayName;
+            }
+
+            Data = httpUrl.Truncate(SchemaConstants.RemoteDependencyData_Data_MaxLength);
+            Target = httpTagObjects.GetHttpDependencyTarget().Truncate(SchemaConstants.RemoteDependencyData_Target_MaxLength);
```

Comment:
TODO: We need to change the logic for target as well.
`GetHttpDependencyTarget`

Reply:
Good catch, yes need to use - GetNewSchemaHttpDependencyTarget. Will cover in the other PR.

Reply:
Also update `ResultCode`